packages/location-management/src/components/EditLocationUnit/index.tsx

```diff
@@ -144,7 +142,44 @@ const EditLocationUnit = (props: EditLocationUnitProps) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [locId]);
 
-  if (loading) {
+  const locationUnits = useQuery(
+    LOCATION_UNIT_FIND_BY_PROPERTIES,
+    () => getBaseTreeNode(opensrpBaseURL),
+    {
+      onError: () => sendErrorNotification(lang.ERROR_OCCURRED),
+      select: (res: LocationUnit[]) => res,
+    }
+  );
+
+  useQueries(
```

Comment:
wouldn't it be better to return the results and do the diffing in a useEffect? That way react-query caches the results and you only have to update state if the results actually change?

Reply:
i tried it.. the way the useQueries(multiple queries) work is different from use query and if was to but the usequery returned variable in the dependency array it the useeffect was running indefinitely continuously so i order to deal with such behavior i had to me use local state that could benefit from react query cache as well as that allow us to easily manage state i come up with this solution

Reply:
I guess the real problem is that useEffect doesn't do deep comparison for dependency array. for which i think we can use [useDeepCompareEffect](https://github.com/kentcdodds/use-deep-compare-effect). See snippet below:

`  const queryResp = useQueries(
    locationUnits.data
      ? locationUnits.data.map((location) => {
          return {
            queryKey: [LOCATION_HIERARCHY, location.id],
            queryFn: () => new OpenSRPService(LOCATION_HIERARCHY, opensrpBaseURL).read(location.id),
            onError: () => sendErrorNotification(lang.ERROR_OCCURRED),
            // Todo : useQueries doesn't support select or types yet https://github.com/tannerlinsley/react-query/pull/1527
            select: (res) => generateJurisdictionTree(res as RawOpenSRPHierarchy).model,
          };
        })
      : []
  );

  useDeepCompareEffect(() => {
    if (queryResp) {
      queryResp.map((res: any) => {
        // the Tree is not already in the locationtree
        if (!treeData.find((data) => JSON.stringify(data) === JSON.stringify(res))) {
          // if the tree already exist i.e only a part of tree is changed, we need to replace old one else add new one
          const alreadyExist = treeData.findIndex(
            (tree) => tree.id === (res as ParsedHierarchyNode).id
          );

          if (alreadyExist !== -1) treeData.splice(alreadyExist, 1, res as ParsedHierarchyNode);
          else setTreeData((treeData) => [...treeData, res as ParsedHierarchyNode]);
        }
      });
    }
  }, [queryResp]);`

Reply:
awesome Let me have a look at it right now

Reply:
https://github.com/opensrp/web/pull/645